### ISSUE TITLE
Can we add an ENV variable configuration has an option bloop client here?

### ISSUE THREAD

**Issue by davidmlw** [sifive/api-scala-sifive#32]
I started a bloop server process and change this line to
`cmd = ["bloop"] + arglist`

It seems working well to do incremental compilation, are there potential problems? Can we use an env var  to select Bloop.cli or a bloop client? Bloop.cli is for CI process and bloop client for user daily work.

https://github.com/sifive/api-scala-sifive/blob/6a298369bb8a1b3543a57f27edc75a52612112d0/lib/scalaplugin.py#L67

**Comment by davidmlw** [comment]
No, it is not OK, as bloop client didn't get the classes out, so no jar out.

**Comment by davidmlw** [comment]
it is not ok, jar is broken. Also need modify make_jar.

**Comment by jackkoenig** [comment]
Also the fundamental problem described in https://github.com/sifive/api-scala-sifive/issues/30#issuecomment-602849318 remains, it doesn't work on multi-user systems and leaves a security hole.

**Comment by jackkoenig** [comment]
I have a branch (that I need to update now that I've extracted the Wit Scala plugin into this repo) that supports incremental compilation via Mill (https://github.com/sifive/api-scala-sifive/tree/mill-incremental-compile). Mill uses local _per-project_ build servers that will give incremental compile. I still want to use Bloop in the long term, but this seems like the easiest path to incremental compile.

**Comment by davidmlw** [comment]
By the way, the bloop server can support multi-user now, by each starting a server with different port number. It should be ok, if the user is a CI process, it can use the same server and port for every task; if the users are human on a same machine, they can allocate the port numbers.

I am not sure the security problem, from my experiments, user A cannot access user B's bloop server.

**Comment by jackkoenig** [comment]
The multiple ports is okay if we wrote the code to manage it which wouldn't be that big of a deal. My main issue is that user A **can** access user B's bloop server. I just tested it and was able to confirm that a colleague can run compilation in my workspace. This opens the door to privilege escalation (note that bloop has a "run" feature so someone could run commands in my workspace as me and write files or do whatever). This may be acceptable to many, in particular if you are not on a multi-user system, but I'm not down to distribute code that has such a glaring vulnerability implicit in its use.
